index.rst

```diff
@@ -202,18 +202,33 @@ File Processing
 
 When a filename is given to EditorConfig a search is performed in the
 directory of the given file and all parent directories for an EditorConfig
-file (named ".editorconfig" by default). Non-existing directories are treated
-as if they exist and are empty. All found EditorConfig files are
-searched for sections with section names matching the given filename. The
-search shall stop if an EditorConfig file is found with the ``root``
-key set to ``true`` in the preamble or when reaching the root
+file. An EditorConfig file is named ".editorconfig", all lowercased.
+Non-existing directories are treated as if they exist and are empty. All found
+EditorConfig files are searched for sections with section names matching the
+given filename. The search shall stop if an EditorConfig file is found with
+the ``root`` key set to ``true`` in the preamble or when reaching the root
 filesystem directory.
 
 Files are read top to bottom and the most recent rules found take
 precedence. If multiple EditorConfig files have matching sections, the rules
 from the closer EditorConfig file are read last, so pairs in closer
 files take precedence.
 
+Capitalization of the File Name
+-------------------------------
+
```

Comment:
I understand why you have added this section, and I think it is good to include!  I would like to suggest we swap the paragraphs so the core behaviour comes first.  How about this?---

```rst
As noted above, ``.editorconfig`` files are named in all lower-case.  
Cores must not check whether the filesystem storing the ``.editorconfig`` 
file is case-sensitive.

*(informative)* For example, on a case-insensitive filesystem, the core
might open ``.editorconfig``, and the OS might return a file
called ``.editorConfig``.  The core would still process the file as usual.  
This way cores do not have to specifically check the case of the filename, 
which can be relatively expensive in the context of EditorConfig.
```

Reply:
I think it's a good idea to swap the paragraphs and make the examples informative. But I think it's too vague to say "Cores must not check whether the filesystem storing the ``.editorconfig``  file is case-sensitive."

I updated the PR to retain the more verbose version of the definition, and move the explanations to a separate informative paragraph. Could you take a look?